config/constants.hpp

```diff
@@ -17,11 +17,28 @@
 #include "micras/proxy/storage.hpp"
 
 namespace micras {
+/*****************************************
+ * Constants
+ *****************************************/
+
 constexpr uint8_t  maze_width{16};
 constexpr uint8_t  maze_height{16};
 constexpr float    cell_size{0.18};
 constexpr uint32_t loop_time_us{1042};
 
+/*****************************************
+ * Template Instantiations
+ *****************************************/
+
+namespace nav {
+using Maze = TMaze<maze_width, maze_height>;
+using Mapping = TMapping<maze_width, maze_height>;
```

Comment:
vc tem alguma referência disso ser um padrão em algum lugar?

n sei se eu gosto disso aqui aisufhasiuf mas se for um jeito comum de fazer essas coisas ok, pq tambem acho feio ficar usando o template pra coisas estaticas e tipos

Reply:
Isso realmente é algo que vale uma justificativa melhor, em resumo, isso não é um padrão que existe, eu inventei ele do 0 e não tenho nenhuma referência pra isso kkkkk. Mas a principal motivação não é simplesmente economizar algumas letra quando chama o template, é uma questão de informação duplicada. No caso, o a quantidade de leds rgb que tem no robô é uma informação, e portanto, seria legal que ela estivesse armazenada em um único lugar, assim como o tamanho do labirinto e as outras coisas passadas nesses templates. Esse using centraliza o local onde essa informação ta armazenada. A alternativa seria definir constantes, assim como foi feito com o tamanho do labirinto, mas ai realmente vira uma questão de economizar letras, pq a gente não ta falando mais de um <2> e sim de um <number_of_leds> que é economizado toda vez.

Dito isso, eu estava bem inclinado a ter esse using no código, a questão que surgiu foi, algum dos nomes teria que ser sacrificado, pq eles não podem ser o mesmo, então o tipo no código que a gente, ou o tipo no código da lib teria que ter um prefixo/sufixo. A única coisa que achei minimamente aceitável foi isso ai, que não me parece um absurdo pq a gente coloca um I antes do nome das interfaces, que é um padrão que veio do C#, então um prefixo T pra template não me parece o fim do mundo.